rdr_service/api/ancillary_study_cloud_tasks_api.py

```diff
@@ -171,25 +172,21 @@ def post(self):
         workflow.execute_workflow()
 
 
-class InsertNphIncidentTaskApi(BaseAncillaryTaskApi):
+class NphIncidentTaskApi(BaseAncillaryTaskApi):
 
     """
-    Cloud Task endpoint: Inserts an incident into a Nph Incident table
-    Mandatory Fields are:
-        dev_note: i.e. "Created a New Incident"
-        message: i.e. "A New Incident"
-        notification_date: i.e. "2023-02-07T13:28:17.239+02:00"
-    Optional Fields are:
-        event_id
-        participant_id
-        src_event_id
-        trace_id
+    Cloud Task endpoint: Can insert an incident into a Nph Incident table.
+    Also send a slack alert to rdr-nph-alerts if the payload sent by partner
+    contains information about participants who have withdrawn from AOU program.
     """
     def post(self):
-        super(InsertNphIncidentTaskApi, self).post()
-        log_msg = f'Insert a new incident with {self.data} for ' \
-                  f'PID: {self.data.get("participant_id")}'
+        super().post()
+        withdrawn_pids = self.data.get("withdrawn_pids")
+        log_msg = f"{len(withdrawn_pids)} withdrawn PIDs received in NPH intake payload."
         logging.info(log_msg)
-        json_payload: Dict[str, Any] = self.data
-        create_nph_incident(save_incident=True, slack=True, **json_payload)
+
+        ts = CLOCK.now().strftime('%Y-%m-%d %H:%M:%S')
+        msg = (f"NPH Intake API Payload received on {ts} contains {len(withdrawn_pids)} withdrawn participant IDs:"
+               f" {', '.join(withdrawn_pids)}")
+        create_nph_incident(slack=True, message=msg)
         return {"success": True}
```

Comment:
So this is super specific to Participant Ids being withdrawn. Can we make this more general as it was before and either create a new cloud task endpoint just for withdrawn actions?